cpu/Makefile.include.cortexm_common

```diff
@@ -58,13 +58,20 @@ export CFLAGS += -DCPU_ARCH_$(ARCH)
 
 # set the compiler specific CPU and FPU options
 ifeq ($(CPU_ARCH),cortex-m4f)
-# TODO: enable hard floating points for the M4F once the context save/restore
-#       code is adjusted to take care of FPU registers
-#export CFLAGS_FPU += -mfloat-abi=hard -mfpu=fpv4-sp-d16
-export MCPU := cortex-m4
-endif
+    USEMODULE += cortexm_fpu
+    ifneq (,$(filter cortexm_fpu,$(DISABLE_MODULE)))
+        export CFLAGS_FPU ?= -mfloat-abi=soft
+    else
+        # clang assumes there is an FPU
+        ifneq (llvm,$(TOOLCHAIN))
```

Comment:
It's there any reason to not give these flags to clang, regardless of the default setting? Does clang give an error if you specify hard float? Toolchains are usually quite configurable and the defaults may not always be the same.